src/services/clipService.ts

```diff
@@ -220,3 +323,119 @@ export async function computeTextEmbedding(
         throw err;
     }
 }
+
+export async function computeONNXTextEmbedding(
+    text: string
+): Promise<Float32Array> {
+    try {
+        const imageSession = await getOnnxTextSession();
+        const tokenizer = getTokenizer();
+        const tokenizedText = Int32Array.from(tokenizer.encodeForCLIP(text));
+        const feeds = {
+            input: new ort.Tensor('int32', tokenizedText, [1, 77]),
+        };
+        const results = await imageSession.run(feeds);
+        const embedVec = results['output'].data; // Float32Array
+        return normalizeEmbedding(embedVec);
+    } catch (err) {
```

Comment:
Instead of erring out, can we check if the model has been downloaded the first thing in this method?

Reply:
the `getOnnxTextSession` calls `getClipTextModelPath` which does this, validates the local downloaded file size (if present) 

or am i understanding it wrong?

Reply:
Then will the error be a model download pending one? That's what you are checking below